### PR TITLE
New Premiere Pro CC Cheat Sheet

### PR DESCRIPTION
**Pull request by JerrinThomas**
## Description of new Instant Answer, or changes

https://duck.co/ia/view/premiere_pro_cc_cheat_sheet

## Testing & Review
To be completed by Community Leader (or DDG Staff) when reviewing Pull Request

**Pull Request**
- [ ] Title follows correct format (Specifies Instant Answer + Purpose)
- [ ] Description contains a valid Instant Answer Page Link (e.g. https://duck.co/ia/view/my_ia)

**Instant Answer Page** (for new Instant Answers)
- [ ] Instant Answer page is correctly filled out and contains:
    - The **Title** is appropriately named and formatted
    - The **IA topics** are present and appropriate
    - The **Description** is clear and coherent
    - **Source Name** exists if applicable
    - All **Example Queries** trigger on [Beta](https://beta.duckduckgo.com/)

**Code**
- [ ] Adheres to the [DuckDuckGo Style Guide](https://docs.duckduckhack.com/resources/code-style-guide.html)
- [ ] Behaviour is appropriately tested. If improvement, tests are adequately extended.
- [ ] There is no unnecessary files in place (such as editor config files)
- [ ] There is no API keys / secrets present
- [ ] Tests are passing (run `$ duckpan test <goodie_id>`)
    - Tester should report any failures

**Ready to merge?**

- [ ] Has this IA been deployed to and tested on [beta.duckduckgo.com](https://beta.duckduckgo.com/)?
- [ ] For larger commits, has this been approved be more than one community member?
- [ ] The number of reviews is appropriate for this type of PR
- [ ] The commit message is clear, coherent and fitting

**Pull Request Review Guidelines**: https://docs.duckduckhack.com/programming-mission/pr-review.html

---

Instant Answer Page: https://duck.co/ia/view/premiere_pro_cc_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: share/goodie/cheat_sheets/json/premierepro.json [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
+    "section_order": [
+        "File",
+        "Edit",
+        "Clip",
+        "Sequence",
+        "Marker",
+        "Window",
+        "Help",
+        "Panels",
+        "Tools",
```
Comment (kirkins) [inline share/goodie/cheat_sheets/json/premierepro.json]: Remove this extra `,`

File: share/goodie/cheat_sheets/json/premierepro.json [hunk 2]
```diff
@@ -0,0 +1,393 @@
...
+                "val": "Pen Tool"
+            },
+            {
+                "key": "H",
+                "val": "Hand Tool"
+            },
+            {
+                "key": "Z",
+                "val": "Zoom Tool"
+            },
```
Comment (kirkins) [inline share/goodie/cheat_sheets/json/premierepro.json]: Remove this `,` too

File: share/goodie/cheat_sheets/json/premierepro.json [hunk 3]
```diff
@@ -0,0 +1,393 @@
...
+                "val": "Match Frame"
+            },
+            {
+                "key": "[Shift] [R]",
+                "val": "Reverse Match Frame"
+            },
+            {
+                "key": "[Ctrl] [K]",
+                "val": "Add Edit"
+            }
```
Comment (kirkins) [inline share/goodie/cheat_sheets/json/premierepro.json]: Missing `,` here

**Comment by JerrinThomas** [comment]
All Errors have been corrected and validated using jsonlint.com

File: share/goodie/cheat_sheets/json/premierepro.json [hunk 4]
```diff
@@ -0,0 +1,393 @@
+{
+    "id": "premiere_cheat_sheet",
```
Comment (kirkins) [inline share/goodie/cheat_sheets/json/premierepro.json]: The ID needs to match the file name.

Add an underscore to the filename `premiere_pro.js`. Then change the ID to `premiere_pro_cheat_sheet`.

Comment (JerrinThomas) [inline share/goodie/cheat_sheets/json/premierepro.json]: name has been changed to premiere.json

File: share/goodie/cheat_sheets/json/premiere.json [hunk 5]
```diff
@@ -0,0 +1,393 @@
+{
+    "id": "premiere_cheat_sheet",
+    "name": "Adobe Premiere Pro CC Cheat Sheet",
+    "description": "Video Editor Premiere Pro CC",
+    "metadata": {
+        "sourceName": "PremiereCheatSheet"
+    },
```
Comment (kirkins) [inline share/goodie/cheat_sheets/json/premiere.json]: Let's add some aliases. As is it is triggered by "Premiere cheat sheet" but not "adobe Premiere cheat sheet" or "Premiere pro cheat sheet".

Add something like this here:

    "aliases": ["premiere pro", "adobe premiere""],

Comment (JerrinThomas) [inline share/goodie/cheat_sheets/json/premiere.json]: added aliases

**Comment by kirkins** [comment]
LGTM 👍

File: share/goodie/cheat_sheets/json/premiere.json [hunk 6]
```diff
@@ -0,0 +1,394 @@
+{
+    "id": "premiere_cheat_sheet",
+    "name": "Adobe Premiere Pro CC Cheat Sheet",
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/premiere.json]: Lets remove 'Cheat sheet' from name

File: share/goodie/cheat_sheets/json/premiere.json [hunk 7]
```diff
@@ -0,0 +1,394 @@
...
+                "val": "Hand Tool"
+            },
+            {
+                "key": "Z",
+                "val": "Zoom Tool"
+            }
+        ],
+        "Clip": [
+            {
+                "key": "[ctrl] [R]",
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/premiere.json]: Use capital c for ctrl

**Comment by mintsoft** [comment]
@JerrinThomas Are you OK with the minor changes here? I'd love to see this go live!

**Comment by sahildua2305** [comment]
> P.S. you can check that it is valid here https://jsonlint.com

@kirkins or we now have an [in-house IA](https://duckduckgo.com/?q=json+validator&atb=v73-4_q&ia=answer) for that 😉

**Comment by mintsoft** [comment]
I'm closing this for now due to inactivity

**Pull request closed without merging**